### ISSUE TITLE
Please remove override of CRM_Core_Payment

### ISSUE THREAD

**Issue by mattwire** [donordepot/com.donordepot.authnetecheck#20]
We've had to disable this extension from automated distribution in CiviCRM because it's causing fatal errors on 4.7.23+ installs.  This is because it is overriding CRM/Core/Payment.php which is no longer allowed for extensions.  Please update your extension so it does not override any core files and request a review so we can get it approved again.

**Comment by sleewok** [comment]
This is probably the cause of my issue (https://github.com/donordepot/com.donordepot.authnetecheck/issues/19)

**Comment by universalhandle** [comment]
Minor point of clarification: overriding core files is allowed for extensions (though discouraged), but it is not allowed for extensions that users can one-click install (i.e., automated distribution).

Since the original release of this extension, the CiviCRM community has published an [extensions lifecycle document](https://docs.civicrm.org/dev/en/latest/extensions/lifecycle/) which outlines some new expectations and guidelines around the use and publication of extensions.

**Comment by zowehlife** [comment]
We would love to use this plugin. Is the override of the payment.php file actually a problem crashing CiviCRM OR is it simply using this projects file instead? IF you are updating the project, what is the ETA?

**Comment by calonmerc** [comment]
Initial work on this fix is hosted on this fork:
https://github.com/calonmerc/com.donordepot.authnetecheck

**Comment by ryeradio** [comment]
I just added the 1.1.3 version, is this a stable version? When I enabled this version I was able to see the check fields, but when I filled out a form and used the check fields I don't see the transaction in my bank.

**Comment by JoshEpic** [comment]
@ryeradio I would use the repo here it seems to be more stable. https://github.com/calonmerc/com.donordepot.authnetecheck

**Comment by ryeradio** [comment]
That is the version I am using, there was no area to post comments on that thread. Do you know if that version is reliable? Again I tried to use this and the transaction did not go through.

**Comment by JoshEpic** [comment]
@ryeradio Try using this release. I had this working for a client but i do not recall what version of civi they were using. https://github.com/donordepot/com.donordepot.authnetecheck/releases/tag/1.2.2

**Comment by mattwire** [comment]
Both this one and the fork have issues that WILL break CiviCRM on anything newer than 4.7.23.  You should be running at least CiviCRM 5.3 otherwise your site has known security vulnerabilities so this extension should not be used.

**Comment by sleewok** [comment]
@mattwire do you have any insight on the specific issue with the fork that will cause it to break? We are trying to squeeze in time to get this working. See the "generic" issue here: https://github.com/calonmerc/com.donordepot.authnetecheck/issues/1

Any help from anyone would be welcome.

**Comment by mattwire** [comment]
@sleewok Yes. CRM_Core_Payment class is fundamental to payments within CiviCRM and changes regularly.  This extension is overriding (replacing) that file and that will cause all kinds of things to break.  The extension needs to stop shipping that file and find another way to perform whatever customisations it's doing.

**Comment by guyiac** [comment]
Given that Donor Depot is no longer supporting this extension, is anyone else interested in taking this on, perhaps as part of a team? I have a client who needs this extension and wants to add capability to process EFT's on the back end and could put some resources towards this.

**Comment by midnightspenguin** [comment]
I agree with Guyiac - I have someone who needs this as well. What would it take to get this extension back on track?

**Comment by guyiac** [comment]
@midnightspenguin my client Asylum Hill Congregational Church has funded some work by Matt Wire to bring this up to snuff. See here: https://github.com/mattwire/com.donordepot.authnetecheck
